src/huggingface_hub/_inference_endpoints.py

```diff
@@ -279,7 +280,8 @@ def update(
             custom_image (`Dict`, *optional*):
                 A custom Docker image to use for the Inference Endpoint. This is useful if you want to deploy an
                 Inference Endpoint running on the `text-generation-inference` (TGI) framework (see examples).
-
+            secrets (`Dict`, *optional*):
+                A dictionary of secrets to be used for building the custom image.
```

Comment:
```suggestion
            secrets (`Dict[str, str]`, *optional*):
                Secret values to inject in the container environment.
```

not only in custom image but any inference endpoint supports them